sync/metrics.go

```diff
@@ -11,40 +12,140 @@ import (
 var meter = otel.Meter("header/sync")
 
 type metrics struct {
-	totalSynced      atomic.Int64
-	totalSyncedGauge metric.Float64ObservableGauge
+	subjectiveHeadInst metric.Int64ObservableGauge
+
+	syncLoopStarted       metric.Int64Counter
+	trustedPeersOutOfSync metric.Int64Counter
+	laggingHeadersStart   metric.Int64Counter
+	readHeader            metric.Int64Counter
+
+	subjectiveHead    atomic.Int64
+	subjectiveHeadReg metric.Registration
+
+	blockTime  metric.Float64Histogram
+	prevHeader time.Time
 }
 
 func newMetrics() (*metrics, error) {
-	totalSynced, err := meter.Float64ObservableGauge(
-		"total_synced_headers",
-		metric.WithDescription("total synced headers"),
+	syncLoopStarted, err := meter.Int64Counter(
+		"hdr_sync_loop_started",
+		metric.WithDescription("sync loop started shows that syncing is in progress"),
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	m := &metrics{
-		totalSyncedGauge: totalSynced,
+	trustedPeersOutOfSync, err := meter.Int64Counter(
+		"hdr_tr_peers_out_of_sync",
+		metric.WithDescription("trusted peers out of sync"),
+	)
+	if err != nil {
+		return nil, err
 	}
 
-	callback := func(ctx context.Context, observer metric.Observer) error {
-		observer.ObserveFloat64(totalSynced, float64(m.totalSynced.Load()))
-		return nil
+	laggingHeadersStart, err := meter.Int64Counter(
```

Comment:
I think laggingHeadersStart is a pretty confusing name. Something like unrecentHeader would make more sense for our codebase. The description would say "tracks every time Syncer returns an unrecent header"